pkg/apis/networking/v1alpha1/clusteringress_types.go

```diff
@@ -93,10 +93,15 @@ type IngressSpec struct {
 	// +optional
 	Rules []ClusterIngressRule `json:"rules,omitempty"`
 
-	// TODO: We need to consider a way to specify if the ClusterIngress address
-	// should be exposed to the Internet, or only exposed privately (cluster local,
-	// VPC, RFC1918).  An example use case is for
-	//   https://github.com/knative/serving/issues/2127.
+	// Visibility setting.
+	Visibility *IngressVisibility `json:"visibility,omitempty"`
+}
+
+// IngressVisibility describes whether the Ingress should be exposed to
+// public gateways or not.
+type IngressVisibility struct {
+	// True iff the ClusterIngress should not be exposed to public gateways.
+	LocalOnly bool `json:"localOnly,omitempty"`
```

Comment:
I think it would be more typical to have IngressVisibility be an enum vs. a bag of bools.

Reply:
Done